SimplenoteIntents/IntentHandler.swift

```diff
@@ -51,13 +43,45 @@ extension IntentHandler: NoteWidgetIntentHandling {
     }
 
     func defaultNote(for intent: NoteWidgetIntent) -> WidgetNote? {
-        guard let dataController = try? WidgetDataController(coreDataManager: coreDataManager) else {
+        guard let note = try? WidgetDataController(context: coreDataManager.managedObjectContext).firstNote() else {
             return nil
         }
 
-        guard let note = try? dataController.firstNote() else {
-            return nil
-        }
         return WidgetNote(identifier: note.simperiumKey, display: note.limitedTitle)
     }
 }
+
+extension IntentHandler: ListWidgetIntentHandling {
+    func provideTagOptionsCollection(for intent: ListWidgetIntent, with completion: @escaping (INObjectCollection<WidgetTag>?, Error?) -> Void) {
+        guard let dataController = try? WidgetDataController(context: coreDataManager.managedObjectContext) else {
+            completion(nil, WidgetError.appConfigurationError)
+            return
+        }
+
+        guard let tags = dataController.tags() else {
+            completion(nil, WidgetError.fetchError)
+            return
+        }
+
+        // Return collection to intents
+        let collection = tagNoteInObjectCollection(from: tags)
+        completion(collection, nil)
+    }
+
+    private func tagNoteInObjectCollection(from tags: [Tag]) -> INObjectCollection<WidgetTag> {
+        let allNotesWidgetTag: [WidgetTag] = [WidgetTag(identifier: SimplenoteConstants.allNotesTagIdentifier, display: Constants.allNotesDisplay)]
```

Comment:
Duplicated in L80. May be worth extracting, if possible, as a WidgetTag extension